chainstate/src/detail/transaction_verifier/mod.rs

```diff
@@ -522,6 +459,7 @@ impl<'a, S: BlockchainStorageRead> TransactionVerifier<'a, S> {
         spend_height: &BlockHeight,
         median_time_past: &BlockTimestamp,
         blockreward_maturity: &BlockDistance,
+        utxo_block_undo: &mut BlockUndo,
```

Comment:
This is heavily frawned upon. We don't pass references of mutable variables because it can easily lead to undefined behavior.

The simplest solution here is to return the tx_undo, and then you can construct the undo at the called level.

Reply:
agree